### DISCUSSION TITLE
Default message for required errors

### DISCUSSION THREAD

**Discussion by reharik** [colinhacks/zod#1859] [Q&A] [answered]
Hi, I would like to replace yup with Zod. Yup has a way to create custom error messages, however you set properties, you are not given a function with the context of the message so you can not specify the name of the property in the error message.  ZodErrorMap is far better.  However, I don't see any way to set a default message for a required field.  e.g. a schema property is required and the data has no value for that property.  I would say that for form validation at least this is 80% of the job. Is there some way to create a custom error message similar to ZodErrorMap that would allow this?
Thanks

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?

```ts
const schema = z.object( {
    name: z.string( { required_error: 'Name is required' } ),
} )

const result = schema.safeParse( {} )
!result.success && console.log( result.error.issues[ 0 ].message )
// Name is required
```

or perhaps this?

```ts
const schema = z.object( {
    name: z.string( {
        errorMap ( issue, ctx ) {
            if ( issue.code === 'invalid_type' && ctx.data === undefined )
                return { message: 'Name is required' }
            return { message: ctx.defaultError }
        }
    } ),
} )

const result = schema.safeParse( {} )
!result.success && console.log( result.error.issues[ 0 ].message )
// Name is required
```

**Reply by reharik** [reply]
No actually I would like something like this 
```ts
const schema = z.object( {
    name: z.string(),
} ).required(name: true);

```
Then some way, similar to or using ZodErrorMap to say for all required errors make the message = "${propertyName} is required"

oh Wait, would that second version work in ZodErrorMap?
I think that is what I'm looking for.  I presume that since there is no `z.string().required()`  that the proper way to declare something as required is as I wrote above. 
Thanks.

**Reply by JacobWeisenburger** [reply]
all schemas in zod are required by default.
